fluss-connectors/fluss-connector-flink/src/test/java/com/alibaba/fluss/connector/flink/source/reader/FlinkSourceSplitReaderTest.java

```diff
@@ -268,6 +268,26 @@ void testHandleMixSnapshotLogSplitChangesAndFetch() throws Exception {
         }
     }
 
+    @Test
+    void testNoSubscribedBucket() throws Exception {
+        TablePath tablePath = TablePath.of(DEFAULT_DB, "test-no-subscribe-bucket-table");
+        Schema schema =
+                Schema.newBuilder()
+                        .column("id", DataTypes.INT())
+                        .column("name", DataTypes.STRING())
+                        .build();
+        final TableDescriptor tableDescriptor =
+                TableDescriptor.builder().schema(schema).distributedBy(1).build();
+        createTable(tablePath, tableDescriptor);
+
+        try (FlinkSourceSplitReader splitReader =
+                createSplitReader(tablePath, schema.toRowType())) {
+            // fetch shouldn't throw exception
+            RecordsWithSplitIds<RecordAndPos> records = splitReader.fetch();
+            assertThat(records.nextSplit()).isNull();
```

Comment:
assert `records.nextRecordFromSplit()` is null as well?

Reply:
If assert `records.nextRecordFromSplit()` is null, it will throw `java.lang.NullPointerException: Make sure nextSplit() did not return null before iterate over the records split.  `FlinkRecordsWithSplitIds#nextRecordFromSplit`will check the `currentRecordIterator` is not null, but if no records in the `RecordsWithSplitIds`, the `currentRecordIterator` will be null.

I think check `nextSplit` is enough since if `nextSplit` is null, flink fetcher won't call method `records.nextRecordFromSplit()`